hawkular/client.go

```diff
@@ -45,7 +47,7 @@ func (c client) toHawkular(mts []plugin.Metric) error {
 	mhs := []metrics.MetricHeader{}
 
 	for _, m := range mts {
-		dp, ty := getFloatOnly(m)
+		dp, ty := getFloatAndString(m)
```

Comment:
I feel it would be easier to read if `getFloatAndString` returned an error.  As it is currently implemented the second return value communicates an error if it's set to Metrics.Generic.  I suggest returning a typed error for this condition.